sequence_processing_pipeline/Pipeline.py

```diff
@@ -444,21 +528,21 @@ def get_project_info(self, short_names=False):
     @staticmethod
     def is_mapping_file(mapping_file_path):
         '''
-        Returns true if file is a mapping file.
+        Returns True if file follows basic mapping-file format.
         '''
         try:
-            Pipeline._validate_mapping_file(mapping_file_path)
-            return True
-        except PipelineError as e:
-            # we want to distinguish between a file that is clearly not a
-            # mapping file e.g. sample-file, and a mapping-file that is perhaps
-            # missing a column or has duplicate sample-names.
-            messages = ['duplicate sample-names detected:', 'missing columns:',
-                        'Column names are case-insensitive.']
+            df = pd.read_csv(mapping_file_path, delimiter='\t', dtype=str)
+        except pd.errors.ParserError:
+            return False
+
+        # if the expected subset of columns required for a mapping-file
+        # are present, then consider this a mapping file, even if it's
+        # an invalid one.
+        exp_columns = frozenset({'barcode', 'tm1000_8_tool',
+                                 'extraction_robot', 'pcr_primers'})
 
-            for message in messages:
-                if str(e).startswith(message):
-                    return True
+        if set(df.columns.str.lower()).issuperset(exp_columns):
```

Comment:
Why not do `return set(df.columns.str.lower()).issuperset(exp_columns)`?

Reply:
Point taken! I probably was just preserving the fall-through case.